reconcile/utils/terraform_client.py

```diff
@@ -792,7 +793,7 @@ def validate_db_upgrade(
                     f"Cannot upgrade RDS instance: {resource_name} "
                     f"from {before_version} to {after_version}"
                 )
-                return
+                sys.exit(1)
```

Comment:
The problem with `raise ValueError' is that it will cause terraform_plan to retry to run. https://github.com/app-sre/qontract-reconcile/blob/7b01b603c8e2f58c1fdd0a8e856135d7673bce6d/utils/terraform_client.py#L137

Reply:
then we should specify `no_retry_exceptions` in `@retry(no_retry_exceptions=xxx)`, use a new exception class just for this validation to avoid wrong matches.